scripts/pre-commit

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+err=0
+
+echo "Verifying C code style..."
+
+FORMAT_TOOL=$(make dump-tools | grep FORMAT | cut -d' ' -f2-)
+CFILES=$(git diff --cached --name-only | grep -E ".*\.[hc]" | grep -vf .format-exclude)
+
+err=0
+for file in $CFILES; do
+  # check stashed version only
+  git show :$file | $FORMAT_TOOL -n --Werror || err=1
```

Comment:
Why not immediately `... || exit 1` here?

Reply:
I want to print all formatting errors here. If we exit after first error it may not give enough information. 

Let's say we are adding a file that shouldn't be formatted, but we forgot to add it to `.format-exclude` . Then we try to commit this file, but we also add some other file which also has bad formatting. Then we may got information about first error in file which we expect to be formatted. But we won't know that we forgot to add a file to `.format-exclude` because it isn't reported.